engine/runtime/src/main/java/org/enso/interpreter/node/ClosureRootNode.java

```diff
@@ -86,9 +87,9 @@ public Object execute(VirtualFrame frame) {
       com.oracle.truffle.api.TruffleSafepoint.poll(this);
     }
     Object state = Function.ArgumentsHelper.getState(frame.getArguments());
-    frame.setObject(this.getStateFrameSlot(), state);
+    frame.setObject(this.getStateFrameSlotIdx(), state);
     Object result = body.executeGeneric(frame);
-    state = FrameUtil.getObjectSafe(frame, this.getStateFrameSlot());
+    state = frame.getObject(this.getStateFrameSlotIdx());
```

Comment:
get/setObjectStatic uses static slots, and for such slots Truffle does not track FrameSlotKind, more specifically, there is only FrameSlotKind.Static. This means that for static slots, we would have to keep the information about its kind internally.

Whereas normal slots (that is indexed slots) conform to "standard lifecycle" - They begin with FrameSlotKind.Illegal, from which they go to, e.g., FrameSLotKind.Double and from there to FrameSlotKind.Object in which they stay - this is what we do, e.g., in AssignmentNode.

On the other hand, stateFrameSlotIdx is a special case - it conforms to monadic state slot which is implicitly assigned to every FrameDescriptor, moreover, it seems that in Enso, this will always be of FrameSlotKind.Object. So for this one particular slot, we could initialize it with setObjectStatic and access it via getObjectStatic, but I personally would prefer to keep it simple and use normal slots everywhere, unless we migrate to static slots altogether.